contracts/WhitelistModule.sol

```diff
@@ -0,0 +1,94 @@
+/*
+ * WhitelistModule contract. Copyright © 2020 by Opyn.co .
+ * Author: Opyn
+ */
+pragma solidity 0.6.10;
+
+import {Ownable} from "./packages/oz/Ownable.sol";
+
+/**
+ * @title WhitelistModule
+ * @dev TODO:
+ */
+
+contract WhitelistModule is Ownable {
+    /** TODO: */
+    address public addressBook;
+
+    /** TODO: */
+    mapping(bytes32 => bool) internal _isSupportedProduct;
+
+    /** TODO: */
+    mapping(address => bool) internal _isSupportedCollateral;
+
+    /** TODO: */
+    mapping(address => bool) public isValidOtoken;
+
+    constructor() public {
+        /** TODO: */
+    }
+
+    /**
+     * @notice TODO:
+     * @dev TODO:
+     * @param underlying TODO:
+     * @param strike TODO:
+     * @param collateral TODO:
+     */
+    function isSupportedProduct(
+        address underlying,
+        address strike,
+        address collateral
+    ) external view returns (bool isValid) {
+        /** TODO: */
+        return true;
+    }
+
+    /**
+     * @notice TODO:
+     * @dev TODO:
+     * @param collateral TODO:
+     */
+    function isSupportedCollateral(address collateral) external view returns (bool isValid) {
+        /** TODO: */
+        return true;
+    }
+
+    /**
+     * @notice TODO:
+     * @dev TODO:
+     * @param underlying TODO:
+     * @param strike TODO:
+     * @param collateral TODO:
+     */
+    function whitelistProduct(
+        address underlying,
+        address strike,
+        address collateral
+    ) external onlyOwner {
+        /** TODO: */
+    }
+
+    /**
+     * @notice TODO:
+     * @dev TODO:
+     * @param collateral TODO:
+     */
+    function whitelistCollateral(address collateral) external onlyOwner {
+        /** TODO: */
+    }
+
+    modifier onlyFactory() {
```

Comment:
I think we don't need this, because only the `init` function is called by the factory, and it already has a "Initializer" modifier, so it can only be called once.

We can discuss more during implementation.